src/theory/quantifiers/sygus/sygus_interpol.h

```diff
@@ -0,0 +1,179 @@
+
+  /**
+   * Set up the grammar for the interpol-to-synthesis.
+   *
+   * The user-defined grammar will be encoded by itpGType. In DEFAULT option, it
+   * will set up the grammar from itpGType. And if itpGType is null, it will set
+   * up the default grammar. In ASSUMPTIONS option, it will set up the grammar
+   * by only using the operators from axioms. In CONCLUSION option, it will set
+   * up the grammar by only using the operators from conj. In SHARED option, it
+   * will set up the grammar by only using the operators shared by axioms and
+   * conj. In ALL option, it will set up the grammar by only using the operators
+   * from either axioms or conj.
+   */
+  TypeNode setSynthGrammar(const TypeNode& itpGType,
+                           const std::vector<Node>& axioms,
+                           const Node& conj);
+
+  /**
+   * Make the interpolation predicate.
+   *
+   * The argument name is the name of the interpol-to-synthesis.
+   */
+  Node mkPredicate(const std::string& name);
+
+  /**
+   * Make the sygus conjecture to be synthesis.
+   *
+   * The argument itp is the interpolation predicate.
+   */
+  void mkSygusConjecture(Node itp,
+                         const std::vector<Node>& axioms,
+                         const Node& conj);
+
+  /**
+   * Get the synthesis solution, stored in interpol.
+   *
+   * The argument itp is the interpolation predicate.
+   */
+  bool findInterpol(Expr& interpol, Node itp);
+
+  /** The SMT engine subsolver
+   *
+   * This is a fresh copy of the SMT engine which is used for making calls
+   * especially for interpolation problem. In particular, consider the input:
+   *   (assert A)
+   *   (get-interpol s B)
+   * In the copy of the SMT engine where these commands are issued, we maintain
+   * A in the assertion stack. To solve the interpolation problem, instead of
+   * modifying the assertion stack to remove A and add the sygus conjecture
+   * (exists I. ...), we invoke a fresh copy of the SMT engine and leave the
+   * assertion stack unchaged. This copy of the SMT engine can be further
+   * queried for information regarding further solutions.
+   */
+  std::unique_ptr<SmtEngine> d_subsolver;
+
+  // The logic for the local copy of SMT engine (d_subsolver).
+  LogicInfo d_logic;
+
+  // symbols from axioms and conjecture.
+  std::vector<Node> d_syms;
+  // shared symbols between axioms and conjecture.
+  std::vector<Node> d_symsShared;
+  // free variables created from d_syms.
+  std::vector<Node> d_vars;
+  // variables created from d_syms for formal argument list.
+  std::vector<Node> d_vlvs;
+  // free variables created from d_symsShared.
+  std::vector<Node> d_varsShared;
+  // variables created from d_symShared for formal argument list.
+  std::vector<Node> d_vlvsShared;
+  // types of shared variables between axioms and conjecture.
+  std::vector<TypeNode> d_varTypesShared;
+  // formal argument list of the interpol-to-synthesis.
+  Node d_abvlShared;
```

Comment:
why abvl?

Reply:
you mean why this name? I don't know, I copied from get-abduct

Reply:
Could you say why the name @ajreynol ? Not sure if the name just carries automatically to the interpolation case.

Reply:
abduct bound variable list maybe?

Reply:
Oh, so I should call it ibvl. Cool